### PR TITLE
GPA.transform: honor the scale parameter (was always applying scaled Procrustes)

### PR DESCRIPTION
**Pull request by AlejandroCoronadoN**
`GPA.transform` ignores the `scale` constructor parameter and always applies scaled Procrustes alignment, so `fit(X).transform(X)` does not reproduce `fit_transform(X)` when `scale=False`.

`fit_transform` correctly branches on `self.scale`: it uses `unscaled_procrustes` (translation + rotation/reflection only) when `scale=False`, and `scipy.spatial.procrustes` (which also unit-normalizes) when `scale=True`. `transform` unconditionally calls `scipy.spatial.procrustes`, which re-introduces the scaling the user disabled.

Repro:

```python
import numpy as np, prince

rng = np.random.RandomState(0)
base = rng.normal(size=(6, 2))
X = np.stack([base, base * 5.0])   # true scale ratio 5x

gpa = prince.GPA(scale=False, init="mean", max_iter=5, random_state=0)
ft = gpa.fit_transform(X.copy())
print(np.linalg.norm(ft[1]) / np.linalg.norm(ft[0]))   # 5.0 (correct)

gpa2 = prince.GPA(scale=False, init="mean", max_iter=5, random_state=0)
t = gpa2.fit(X.copy()).transform(X.copy())
print(np.linalg.norm(t[1]) / np.linalg.norm(t[0]))     # was 1.0, now 5.0
```

The fix mirrors the branch already used in `fit_transform`. Added a regression test asserting `fit(X).transform(X)` reproduces `fit_transform(X)` with `scale=False`.

### CODE REVIEW

**Comment by MaxHalford** [comment]
Thanks!
